### PR TITLE
Update temperature pulls 

### PR DESCRIPTION
**Pull request by limnoliver**
This PR does a few things: 
- adds a crosswalk file that summarizes the site to segment data
- makes some small changes to Rasha's WQP inventory change that switched us from using states to years to inventory data. The change ensure we're inventorying up to the pull date, and relies on the pull date to modify the start years so we don't have to manually modify next year.
- fixes site naming that are pulled through from Ecosheds data. I noticed there was no way to go from the site ID we were keeping to their data viewer online, and no way to identify USGS site numbers that may exist in other data sources. 
- Re-pulls both WQP and NWIS data for the temperature visualization. 

Note, during the WQP munge, I hit memory issues. Time to move this pipeline to Denali.

### CODE REVIEW

**Comment by limnoliver** [comment]
Note: 

- WQP inventory increased to 656956 from 656627 sites = +329 sites.
- WQP data increased from 340901 to 369372 = +28471 sites.

This is because Rasha [recently repulled the inventory](https://github.com/USGS-R/2wp-temp-observations/pull/40/files), but the actual data haven't been pulled since 2020-07-29. When Rasha pulled, we had an increase in +38k sites.

File: 1_wqp_pull.yml [hunk 1]
```diff
@@ -41,8 +42,7 @@ targets:
   # only filters by characteristicNames, and pulls all at the same time
 
   start_year:
-    command: c(I('1900'), I(seq(1980, 1997, by = 3)),
-      I(seq(1998, 2003, by = 2)), I(seq(2004, 2020)))
+    command: get_start_years(begin = 1900, end = wqp_pull_date)
```
Comment (hcorson-dosch) [inline 1_wqp_pull.yml:45]: This does seem cleaner to have the end year set from the wqp_pull_date, and to have the sequencing documented in a function.

File: 6_network.yml [hunk 2]
```diff
@@ -60,14 +60,11 @@ targets:
...
 
-  #6_network/out/site_stream_crosswalk.rds.ind:
-   # command: create_crosswalk(
-    #  out_ind = target_name,
-     # matched_ind = '6_network/out/site_flowlines.rds.ind',
-      #sites_ind = '5_data_munge/out/stream_sites_us.rds.ind')
-
-  #6_network/out/stream_site_crosswalk.rds.ind:
-   # command: add_segment_info()
+  6_network/out/site_stream_crosswalk.rds.ind:
```
Comment (hcorson-dosch) [inline 6_network.yml:63]: This seems like a really helpful addition

File: 5_data_munge/src/munge_data_files.R [hunk 3]
```diff
@@ -48,6 +53,12 @@ munge_ecosheds <- function(in_ind, min_value, max_value, out_ind) {
 
   message(nrow(dat_out) - nrow(dat_out2), ' temperature observations dropped due to comments that indicated poor data quality.')
 
+  # get the right site ID in the dataframe
+  # site ID is a combo of agency_name and location_name from Ecosheds
```
Comment (hcorson-dosch) [inline 5_data_munge/src/munge_data_files.R:57]: Nice catch!

**Review by hcorson-dosch — APPROVED** [review]
Sorry for the delay. I did a quick review of your logic for building the start years and your changes to the record summaries (adding the Ecosheds id to the dataframe and adding the site to reach crosswalk file). Everything looks good.

**Pull request merged**